tests/test_gateway.py

```diff
@@ -139,6 +139,93 @@ def test_submit_python_script_with_replicate(self):
         self.assertTrue(isinstance(jobId, numbers.Number))
         self.gateway.disconnect()
 
+    def test_submit_python_script_with_loop(self):
+        self.gateway.connect(self.username, self.password)
+
+        # Creating the split task
+        pythonTaskStart = self.gateway.createPythonTask()
+        pythonTaskStart.setTaskName("StartPythonTask")
+        pythonTaskStart.setTaskImplementation("""print("Hello world!")""")
+
+        pythonTaskStart.setFlowBlock(self.gateway.getProactiveFlowBlockType().start())
+
+        # Creating the replicated task
+        pythonTaskLoop = self.gateway.createPythonTask()
+        pythonTaskLoop.setTaskName("LoopPythonTask")
+        pythonTaskLoop.setTaskImplementation("""print("Hello world!")""")
+        pythonTaskLoop.addDependency(pythonTaskStart)
+
+        loopScript = """
+if(variables.get('PA_TASK_ITERATION') < 5) {
+    loop = true;
+} else {
+    loop = false;
+}"""
+        flow_script = self.gateway.createLoopFlowScript(loopScript, pythonTaskStart.getTaskName())
+        pythonTaskLoop.setFlowScript(flow_script)
+        pythonTaskLoop.setFlowBlock(self.gateway.getProactiveFlowBlockType().end())
+
+        myJob = self.gateway.createJob()
+        myJob.setJobName("SimplePythonJobWithLoopTask")
+
+        myJob.addTask(pythonTaskStart)
+        myJob.addTask(pythonTaskLoop)
+
+        jobId = self.gateway.submitJob(myJob)
+
+        self.assertIsNotNone(jobId)
+        self.assertTrue(isinstance(jobId, numbers.Number))
+        self.gateway.disconnect()
+
+    def test_submit_python_script_with_branch(self):
+        self.gateway.connect(self.username, self.password)
+
+        # Creating the split task
+        pythonTaskCondition = self.gateway.createPythonTask()
+        pythonTaskCondition.setTaskName("ConditionPythonTask")
+        pythonTaskCondition.setTaskImplementation("""print("Hello condition world!")""")
+
+        # Creating the If task
+        pythonTaskIf = self.gateway.createPythonTask()
+        pythonTaskIf.setTaskName("IfPythonTask")
+        pythonTaskIf.setTaskImplementation("""print("If task.")""")
+
+        # Creating the Else task
+        pythonTaskElse = self.gateway.createPythonTask()
+        pythonTaskElse.setTaskName("ElsePythonTask")
+        pythonTaskElse.setTaskImplementation("""print("Else task should be skipped.")""")
+
+        # Creating the Continuation task
+        pythonTaskContinuation = self.gateway.createPythonTask()
+        pythonTaskContinuation.setTaskName("ContinuationPythonTask")
+        pythonTaskContinuation.setTaskImplementation("""print("Continuation task.")""")
+
+        branchScript = """
+if(1==1){
+    branch = "if";
+} else {
+    branch = "else";
```

Comment:
do you really need this else?  1==1 evaluates to true right?

Reply:
No. It does not change anything. It is the implementation of the branching script.